File: IzquierdoAndres_Musica/IzquierdoAndres_Musica/Models/Album.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace IzquierdoAndres_Musica.Models;

public partial class Album
{
    [Key]
    public int AlbumId { get; set; }

    [DisplayName("Título")]
    [Required(ErrorMessage = "Obligatorio")]
    [MaxLength(50, ErrorMessage = "El título no puede tener más de 50 carácteres.")]
    [MinLength(1, ErrorMessage = "El título no puede tener menos de 3 carácteres.")]
    public string Title { get; set; } = null!;

    public int ArtistId { get; set; }

    [DisplayName("Artista")]
    public virtual Artist Artist { get; set; } = null!;
}

File: Java/Spring Boot Exam/src/main/java/es/cifpcm/IzquierdoAndres_SPRING/Models/Album.java
package es.cifpcm.IzquierdoAndres_SPRING.Models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.io.Serializable;

@Entity
@Table(name = "album")
public class Album implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "AlbumId", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer albumId;

    @NotBlank(message = "El título no puede estar vacío.")
    @Size(min = 1, max = 160, message = "El título debe tener al menos un carácter y como máximo 160.")
    @Column(name = "Title", nullable = false)
    private String title;

    @Column(name = "ArtistId", nullable = false)
    private Integer artistId;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "ArtistId", nullable = false, insertable = false, updatable = false)
    private Artist artist;

    public Album(Integer albumId, String title, Integer artistId) {
        this.albumId = albumId;
        this.title = title;
        this.artistId = artistId;
    }

    public Album(){

    }

    public Integer getAlbumId() {
        return albumId;
    }

    public void setAlbumId(Integer albumId) {
        this.albumId = albumId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getArtistId() {
        return artistId;
    }

    public void setArtistId(Integer artistId) {
        this.artistId = artistId;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }
}
